### PR TITLE
fix some warnings

### PR DESCRIPTION
**Pull request by jpfeuffer**
## Description

fix compile warnings

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

### **PR Type**
Bug fix

___

### **Description**
- Removed several unused variables across multiple files to clean up the codebase.
- Fixed a logical operator issue in `TextFile.cpp`.
- Updated loop variable types and included necessary headers for better type safety.
- Removed redundant code and checks to improve code efficiency and readability.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>10 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>IDMapper.cpp</strong><dd><code>Remove unused variable and clean up code</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/ANALYSIS/ID/IDMapper.cpp

<li>Removed unused variable <code>spectrum_matches</code>.<br> <li> Cleaned up code by removing unnecessary increment operation.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-879c02fad465958b5861f34e184e21025828f110c4604271473ec5bb50ad58e1">+0/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>MRMFeatureSelector.cpp</strong><dd><code>Remove unused variables and clean up code</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureSelector.cpp

<li>Removed unused variables <code>n_constraints</code> and <code>n_variables</code>.<br> <li> Cleaned up code by removing unnecessary increment operations.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-ce3a94dfaa8f7b14dee9f6339262527e6b833fd4fcca6c420048304fd78cfc76">+0/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>MSstatsFile.cpp</strong><dd><code>Remove redundant duplicate check and clean up code</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/FORMAT/MSstatsFile.cpp

<li>Removed redundant code for checking duplicate entries.<br> <li> Cleaned up code by removing unnecessary variable <code>count_similar</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-84c58d52d8f0b0f1d9e495d1237c14b81d12927ba9606e3a54a18a90ad989172">+0/-11</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>PercolatorInfile.cpp</strong><dd><code>Remove unused variable</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/FORMAT/PercolatorInfile.cpp

- Removed unused variable `to_idx_t`.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-af44b8fb973ecf4ed877320b459149c50a4514feddada33a87b7b587ebe9a446">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>QcMLFile.cpp</strong><dd><code>Remove unused variable</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/FORMAT/QcMLFile.cpp

- Removed unused variable `runs_count`.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-42bfcbd1901b97171d7e6ac773346858876b4f17304c31b7f3ca4b9e00a6eafe">+0/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>TextFile.cpp</strong><dd><code>Fix logical operator in comment line check</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/FORMAT/TextFile.cpp

- Fixed logical operator from `&` to `&&` for comment line check.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-203728cf2f1b5e44f40cc0ee5c69524edd6eb4e334d19079b2e2730ea030ba98">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>IMDataConverter.cpp</strong><dd><code>Include cstddef and update loop variable type</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/IONMOBILITY/IMDataConverter.cpp

<li>Added <code>#include <cstddef></code> for size_t.<br> <li> Changed loop variable type from <code>int</code> to <code>size_t</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-ef7697b5f8f1b6e5a60dad2e7cd233e31a00ee96465a3d5c3a13ea3b0a9e764a">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ExperimentalDesign.cpp</strong><dd><code>Remove unused variable</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/METADATA/ExperimentalDesign.cpp

- Removed unused variable `fraction_groups_assigned`.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-d41bf19ce8e5285da957d18d0e3239127407a47bd83cf9aaf8bb62ce040a4363">+0/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>DBSuitability.cpp</strong><dd><code>Remove unused variable</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/QC/DBSuitability.cpp

- Removed unused variable `curr_hit`.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-7068a951e296ba7ae2cce4049f630b167690c830a3a71204ed9482cc66712511">+0/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>MQMsmsExporter.cpp</strong><dd><code>Remove redundant loop for peptide identification size</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/QC/MQMsmsExporter.cpp

- Removed redundant loop for calculating `pep_ids_size`.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7654/files#diff-89822cfe77903f3be6806c41cd0cc2afca1b2dacb59b8850a24992ff0e8febf0">+1/-13</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Review by cbielow — APPROVED** [review]

**Review by poshul — APPROVED** [review]

**Comment by jpfeuffer** [comment]
How was this mergeable? There were conflicts. Also I think we should have waited for the tests.

File: src/openms/source/FORMAT/MSstatsFile.cpp [hunk 1]
```diff
@@ -108,8 +108,6 @@ void MSstatsFile::constructFile_(const String& retention_time_summarization_meth
                                          LineType& peptideseq_to_prefix_to_intensities) const
 
 {
-  // sanity check that the triples (peptide_sequence, precursor_charge, run) only appears once
-  set<tuple<String, String, String> > peptideseq_precursor_charge_run;
```
Comment (jpfeuffer) [inline src/openms/source/FORMAT/MSstatsFile.cpp:112]: This here was a conflict, and I prefer the old style.

**Comment by poshul** [comment]
> How was this mergeable? There were conflicts. Also I think we should have waited for the tests.

Err didn't the tests pass above? If I misread that, my bad

**Comment by jpfeuffer** [comment]
They "pass" because they could not be run due to the merge conflict.

**Comment by poshul** [comment]
That's my f-up then, sorry, its been a rough couple days. Should I go ahead and revert it?

**Comment by jpfeuffer** [comment]
I added branch protection rules to develop. Passing actions are required for merge now.

File: src/openms/source/FORMAT/MSstatsFile.cpp (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/source/FORMAT/MSstatsFile.cpp:112]: @timosachsenberg I think the change that was overwritten by the conflict was yours. Can you check if you want to revert this PR or just revert that part somehow?

**Pull request merged**